rust-install/src/manifest/mod.rs

```diff
@@ -240,6 +241,89 @@ impl Manifestation {
         Ok(tx)
     }
 
+    fn read_config(&self) -> Result<Option<Config>> {
+        let prefix = self.installation.prefix();
+        let ref rel_config_path = prefix.rel_manifest_file(CONFIG_FILE);
+        let ref config_path = prefix.path().join(rel_config_path);
+        if utils::path_exists(config_path) {
+            let ref config_str = try!(utils::read_file("dist config", config_path));
+            Ok(Some(try!(Config::parse(config_str))))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Installation using the legacy v1 manifest format
+    pub fn update_v1(&self,
+                     new_manifest: &[String],
+                     update_hash: Option<&Path>,
+                     temp_cfg: &temp::Cfg,
+                     notify_handler: NotifyHandler) -> Result<Option<String>> {
+        // If there's already a v2 installation then something has gone wrong
+        if try!(self.read_config()).is_some() {
+            return Err(Error::ObsoleteDistManifest);
+        }
+
+        let url = new_manifest.iter().find(|u| u.contains(&self.target_triple));
```

Comment:
We should be specifically looking for `target_triple + ".tar.gz"` here, otherwise it will just find whichever file happens to be listed first for that target.